### ISSUE TITLE
Problems with certain regex

### ISSUE THREAD

**Issue by blankster** [mpociot/botman#390]
- BotMan Version: 1.5.6
- PHP Version: 7.0.9
- Messaging Service(s): Telegram
- Cache Driver: Doctrine

### Description:
I don't know the exact limits of regex usage, but I've again and again problem with some more advanced ones. I use them for the ask-function as pattern.

### Steps To Reproduce:
(without ^and $ also doesn't work)
All numbers from 0 to 23: ^([0]?[0-2][0-3]|[0-9])$
All numbers from 0 to 59: ^([0]?[0-5][0-9]|[0-9])$

When I use for example "([0]?[0-2][0-3]|[0-9])" as pattern with the ask-function I get the errror: array_combine(): Both parameters should have an equal number of elements in \vendor\mpociot\botman\src\Mpociot\BotMan\Traits\HandlesConversations.php on line 162

It would be extremely great if anybody could help me to translate these (working) regex to the form I can use them for BotMan. Is the solution something similar like https://github.com/mpociot/botman/issues/76 ?

**Comment by mpociot** [comment]
The start of string / end of string modifiers get added to your regular expression automatically.

This works for me:

```
$botman->hears('([0]?[0-2][0-3]|[0-9])', function ($bot) {
    $bot->reply("foo");
});
```

**Comment by blankster** [comment]
Thanks for your answer!

Is it possible that the behavior of the ask-function with multiple patterns is working differently? (the error message mentions also the conversation part). I didn't try it with a hears-function as it's inside a conversation with multiple patterns and I'm quite sure I've exactly used this pattern (as written above).

My temporary workaround was to catch all values and deal with it inside the closure, but that's of course not really nice.

**Comment by mpociot** [comment]
You're right. There was an issue with the conversation patterns.
It's now fixed, but only available in the next release - which will be 2.0.
Backporting it to the 1.5 branch might take some time, sorry.
